### PR TITLE
Added more advanced control to the Arm subsystem

### LINKED ISSUE #22: Motor Defaults
In the Arm and Claw subsystems, the motors all need to be reset to factory defaults every time the robot is turned on.  Sort of as a precaution to make sure there aren't any weird settings.

Add a method that looks something like this to each of the two subsystems:
```
  private static void configDriveMotor(WPI_TalonFX driveMotor) {
    driveMotor.configFactoryDefault();
    driveMotor.setNeutralMode(NeutralMode.Brake);
    driveMotor.config_kP(0, p);
    driveMotor.config_kI(0, i);
    driveMotor.config_kD(0, d);
    driveMotor.config_kF(0, f);
  }
```
The pidf values we will have to tune when we get a real robot.  For now say 1, 0, 0, 0 for p, i, d, and f respectively.
After this method is added, we should call this method in the constructor for each of the motors initialized in the subsystem.

### PR DESCRIPTION
**Pull request by StephenDay1**
I added several methods to the Arm subsystem to more fully control the arm.

set was renamed to setExtension, as it controls the extending motors

setShoulder controls the lifting of the arm from the shoulder joint.

toPosition is the highest level of control.  Given an x and y coord in the from of a Translation2d, it moves the arm to the desired position

toExtension extends the arm to the desired distance in meters

toAngle lifts the arm to the desired angle above 0, which is when the arm is facing straight down.

There are also 3 getter methods for that do the inverse of the three "to_____" methods.

Please add notes and ask questions, I realize this code is complicated.  The basic idea behind it is just doing math with a triangle.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Claw.java [hunk 1]
```diff
@@ -57,4 +60,13 @@ public void toggleClaw() {
   public void periodic() {
     // This method will be called once per scheduler run
   }
+  private static void configDriveMotor(WPI_TalonFX driveMotor){
+    driveMotor.configFactoryDefault(); // resets the motor to its factory default settings. 
+    driveMotor.setNeutralMode(NeutralMode.Brake); //sets neutral mode of robot to break.
```
Comment (StephenDay1) [inline src/main/java/frc/robot/subsystems/Claw.java:65]: NeutralMode needs to be imported

File: src/main/java/frc/robot/subsystems/Claw.java [hunk 2]
```diff
@@ -57,4 +60,13 @@ public void toggleClaw() {
...
+    driveMotor.config_KP(1, p);
+    driveMotor.config_KI(0, i);
+    driveMotor.config_KD(0, d);
+    driveMotor.config_KF(0, f);
```
Comment (StephenDay1) [inline src/main/java/frc/robot/subsystems/Claw.java:69]: The variables p, i, d, and f were just put there as an example.  They should be changed for the actual numbers, 1, 0, 0, and 0.
Also, the real method call should be spelled with a lowercase k.

File: src/main/java/frc/robot/subsystems/Arm.java [hunk 3]
```diff
@@ -1,30 +1,183 @@
...
+            arm.setLength(getArmExtensionDistance());
+        }
+    }
+    private static void configDriveMotor(WPI_TalonFX driveMotor){
+        driveMotor.configFactoryDefault(); // resets the motor to its factory default settings. 
+        driveMotor.setNeutralMode(NeutralMode.Brake); //sets neutral mode of robot to break.
+        driveMotor.config_KP(1, p);
+        driveMotor.config_KI(0, i);
+        driveMotor.config_KD(0, d);
+        driveMotor.config_KF(0, f);
```
Comment (StephenDay1) [inline src/main/java/frc/robot/subsystems/Arm.java:180]: Same issue as in Claw.java.  See the comment there.

**Review by StephenDay1 — COMMENTED** [review]
@24ginemar1 Can you make the changes that I made comments on?

**Review by 24ginemar1 — APPROVED** [review]

**Pull request merged**